### PR TITLE
Refactor getting various sub options into helper methods

### LINKED ISSUE #177: Refactor Generator code to be DRY and more concise
Currently the code for each generator is very long, hard to follow and there is duplicated logic between generators.

To increase the chance of attracting contributors, the code should be easy for them to follow and reuse where possible. This will lower the barrier to entry for writing new generators.

### PR DESCRIPTION
**Pull request by kylehodgetts**
This PR resolves #177 

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: options/timeout.go [hunk 1]
```diff
@@ -1,12 +1,37 @@
...
+import "reflect"
+
 type TimeoutOptions struct {
 	// RequestTimeout is total request timeout
 	RequestTimeout uint32 `yaml:"request_timeout,omitempty" json:"request_timeout,omitempty"`
 	// IdleTimeout is timeout for idle connection
 	IdleTimeout uint32 `yaml:"idle_timeout,omitempty" json:"idle_timeout,omitempty"`
 }
 
+func (o *Options) GetTimeoutOptions(path, method string) TimeoutOptions {
```
Comment (dobegor) [inline options/timeout.go:12]: I'd use the same naming there as others, `GetTimeoutOpts`

**Review by dobegor — APPROVED** [review]

File: options/timeout.go (hunk 1 above)
Comment (kylehodgetts) [reply options/timeout.go]: Done

**Review by dobegor — APPROVED** [review]
